### ISSUE TITLE
error 1013 :Invalid quatity when selling on binance 

### ISSUE THREAD

**Issue by zezto** [enigmampc/catalyst#448]
Dear Catalyst Maintainers,

Before I tell you about my issue, let me describe my environment:

# Environment
ubuntu 14.0 lts
python 3.6
python bitness: 32
installed catalyst through conda 

* Python packages: `$ pip freeze` or `$ conda list`
# Name                    Version                   Build  Channel
aiodns                    1.1.1                     <pip>
aiohttp                   3.0.1                     <pip>
alembic                   0.9.7                     <pip>
async-timeout             2.0.0                     <pip>
attrdict                  2.0.0                     <pip>
attrs                     17.4.0                    <pip>
backcall                  0.1.0                      py_0    conda-forge
bcolz                     1.2.1                     <pip>
bleach                    2.1.3                      py_0    conda-forge
boto3                     1.5.27                    <pip>
botocore                  1.8.41                    <pip>
Bottleneck                1.2.1                     <pip>
ca-certificates           2017.08.26           h1d4fec5_0
cchardet                  2.1.1                     <pip>
ccxt                      1.12.131                  <pip>
certifi                   2018.1.18                py36_0
chardet                   3.0.4                     <pip>
click                     6.7                       <pip>
contextlib2               0.5.5                     <pip>
cycler                    0.10.0                     py_1    conda-forge
cyordereddict             1.0.0                     <pip>
Cython                    0.27.3                    <pip>
cytoolz                   0.9.0                     <pip>
dbus                      1.13.2               h714fa37_1
decorator                 4.3.0                      py_0    conda-forge
docutils                  0.14                      <pip>
empyrical                 0.2.1                     <pip>
enigma-catalyst           0.5.18                    <pip>
entrypoints               0.2.3                    py36_2
eth-abi                   1.1.1                     <pip>
eth-account               0.2.2                     <pip>
eth-hash                  0.1.3                     <pip>
eth-keyfile               0.5.1                     <pip>
eth-keys                  0.2.0b3                   <pip>
eth-rlp                   0.1.2                     <pip>
eth-utils                 1.0.3                     <pip>
expat                     2.2.5                h8d48cb0_0
fontconfig                2.12.6               h49f89f6_0
freetype                  2.8                  hb4d7b56_1
glib                      2.56.1               h000015b_0
gmp                       6.1.2                h8afb2fd_1
gst-plugins-base          1.14.0               hbbd80ab_1
gstreamer                 1.14.0               hb453b48_1
hexbytes                  0.1.0                     <pip>
html5lib                  1.0.1                      py_0    conda-forge
icu                       58.2                 he175447_1
idna                      2.6                       <pip>
idna-ssl                  1.0.0                     <pip>
intel-openmp              2018.0.0                      8
intervaltree              2.1.0                     <pip>
ipykernel                 4.8.2                    py36_0
ipython                   6.4.0                    py36_1
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.0.3                      py_0    conda-forge
jedi                      0.12.0                   py36_1
jinja2                    2.10                       py_1    conda-forge
jmespath                  0.9.3                     <pip>
jpeg                      9b                   h29c5dd3_2
jsonschema                2.6.0            py36had8cd67_0
jupyter                   1.0.0                      py_1    conda-forge
jupyter_client            5.2.3                      py_1    conda-forge
jupyter_console           5.2.0                    py36_1
jupyter_core              4.4.0                      py_0    conda-forge
kiwisolver                1.0.1            py36hf484d3e_0
libedit                   3.1.20170329         h6b74fdf_2
libffi                    3.2.1                h97ff0df_4
libgcc-ng                 7.2.0                h9268252_3
libgfortran-ng            7.2.0                h9268252_3
libpng                    1.6.34               hb9fc6fc_0
libsodium                 1.0.16               h1bed415_0
libstdcxx-ng              7.2.0                h9268252_3
libxcb                    1.13                 h1bed415_1
libxml2                   2.9.8                hf84eae3_0
Logbook                   0.12.5                    <pip>
lru-dict                  1.1.6                     <pip>
lxml                      4.1.1                     <pip>
Mako                      1.0.7                     <pip>
markupsafe                1.0              py36h14c3975_1
matplotlib                2.2.2            py36h0e671d2_1
mistune                   0.8.3                      py_0    conda-forge
mkl                       2018.0.1             h4fb54b9_4
multidict                 4.1.0                     <pip>
multipledispatch          0.4.9                     <pip>
nbconvert                 5.3.1                      py_1    conda-forge
nbformat                  4.4.0                      py_1    conda-forge
ncurses                   6.1                  hf484d3e_0
networkx                  2.1                       <pip>
notebook                  5.5.0                    py36_0
numexpr                   2.6.4                     <pip>
numpy                     1.14.0           py36ha266831_2
numpy                     1.15.1                    <pip>
openssl                   1.0.2n               hb7f436b_0
pandas                    0.19.2                    <pip>
pandas-datareader         0.6.0                     <pip>
pandoc                    1.15.0.6             h57d12dc_1
pandocfilters             1.4.2                      py_1    conda-forge
parsimonious              0.8.0                     <pip>
parso                     0.3.0                      py_0    conda-forge
patsy                     0.5.0                     <pip>
pcre                      8.42                 h439df22_0
pexpect                   4.6.0                    py36_0
pickleshare               0.7.4            py36hef8e16a_0
pip                       9.0.1                    py36_5
prompt_toolkit            1.0.15           py36h1d3e51a_0
ptyprocess                0.6.0                    py36_0
pycares                   2.3.0                     <pip>
pycryptodome              3.6.1                     <pip>
pygments                  2.2.0                      py_1    conda-forge
pyparsing                 2.2.0                      py_1    conda-forge
pyqt                      5.9.2            py36h751905a_0
pysha3                    1.0.2                     <pip>
python                    3.6.4                hc3d631a_3
python-dateutil           2.7.3                      py_0    conda-forge
python-editor             1.0.3                     <pip>
pytz                      2018.5                     py_0    conda-forge
pytz                      2016.4                    <pip>
pyzmq                     17.0.0           py36h14c3975_3
qt                        5.9.5                h7e424d6_0
qtconsole                 4.3.1            py36h59098de_0
readline                  7.0                  h6b4b497_4
redo                      1.6                       <pip>
requests                  2.18.4                    <pip>
requests-file             1.4.3                     <pip>
requests-ftp              0.3.1                     <pip>
requests-toolbelt         0.8.0                     <pip>
rlp                       0.6.0                     <pip>
s3transfer                0.1.12                    <pip>
scipy                     1.0.0            py36h5a0cb3a_0
send2trash                1.5.0                      py_0    conda-forge
setuptools                38.4.0                   py36_0
setuptools                38.5.1                    <pip>
simplegeneric             0.8.1                      py_1    conda-forge
sip                       4.19.8           py36hf484d3e_0
six                       1.11.0                   py36_1
sortedcontainers          1.5.9                     <pip>
SQLAlchemy                1.2.2                     <pip>
sqlite                    3.22.0               h1bed415_0
statsmodels               0.8.0                     <pip>
TA-Lib                    0.4.17                    <pip>
tables                    3.4.2                     <pip>
terminado                 0.8.1                    py36_1
testpath                  0.3.1            py36h69249d3_0
tk                        8.6.7                h9a736a7_3
toolz                     0.9.0                     <pip>
tornado                   5.0.2            py36h14c3975_0
traitlets                 4.3.2            py36h6ad27bd_0
urllib3                   1.22                      <pip>
virtualenv                15.2.0                    <pip>
wcwidth                   0.1.7                      py_1    conda-forge
web3                      4.2.1                     <pip>
webencodings              0.5.1                    py36_1
websockets                5.0                       <pip>
wheel                     0.30.0           py36h89e8202_1
widgetsnbextension        3.2.1                    py36_0
wrapt                     1.10.11                   <pip>
xz                        5.2.3                h5e939de_4
yarl                      1.1.0                     <pip>
zeromq                    4.2.5                hf484d3e_0
zlib                      1.2.11               h907e355_2

Now that you know a little about me, let me tell you about the issue I am
having:
When I place use dual moving average script it buys when their a crossover it buys at the current price in which the crossover happens. when a negative crossover happens it sells at the current price. But once it sells I get an error right away 
WARNING: CCXT: the exchange rejected the order: binance {"code":-1013,"msg":"Invalid quantity."}
I suspect its catalyst trying to sell something less than .001 btc (which is the minimum sale price on binance. Is there something I can modify to avoid this ? or place the sell all on one order?
`$ whoami`
- zezto

**Comment by lenak25** [comment]
@zezto, sorry for the very late response. 

The examples are intended to help users to easily start with catalyst and to provide a reference for defining and running algorithms. Before starting with live trading, we encourage users to test their algorithm both in backtesting (if applicable) and paper trading.

In regards to your specific question, this indeed seems an issue providing a small quantity. And I agree with @FWangTrading comment.

@zezto , I am closing this, but feel free to reopen/open a new issue if you still experience problems.
